### ISSUE TITLE
[FAQ] State of Fastload and the future of it. 

### ISSUE THREAD

**Issue by KirboSoftware** [BumbleSoftware/Fastload#115]
👋 Good day to you all, Kirbo here, the co-owner of this project. 

Since Bumble (StockiesLad) left the project and all stuff relating to it and newer mc versions coming out, I plan out to pick up Fastload again and rework the project to make it less pain to maintain for whoever desires to pick this up. 

## What will change?
Multiloader support. Bumble aka StockiesLad made Fastload rather abstract from mc versions, which in term is alright, but means it would not really work when updating this for forge, neo etc (since this was a fabric first mod) and updating this to newer versions is also a bit of pain. 

So I will personally spend some time to rewrite the project to support all modloaders in a single branch and pray it just works.. 

## Who will continue to work on this?
It will probably be done soley by me now, but if other people are interested in helping me maintaining this, I wouldn't say no to it 👍 

## Neoforge support?
Yes, I plan to support NeoForge down the line when I've made the multiloader side of fastload work. (this is a thing for 1.20.4+)

## Future mc versions?
I do my best to keep it up to date with the latest changes and make it compatible with every mod out there, but since this will be most likely be done by me now, it will take time. So I will ask for patience. 

## Where will the new multiloader repo be?
Probably it will have it's own repo since I gotta rewrite the mod from scratch and would keep me sane while developing. 

## Bugfixes?
Yup, I do my best to tackle all issues relating to the mod and all compat issues relating to it. 

## Will this project be abandoned?
Probably not, until Mojang will implement this fully themselves, I will do my best to keep the project alive 👍 and also all help is welcome keeping the mod alive. I'm open for people to help with this project and try to revive it from the dead.

**Comment by LostLuma** [comment]
> Future mc versions?

I don't think FastLoad is going to be very useful in versions past 1.20.5, since the spawn chunks are now only an area of 3x3 entity ticking chunks by default instead of the 19x19 before, and players can also just disable spawn chunks altogether without any mods.

**Comment by KirboSoftware** [comment]
> and players can also just disable spawn chunks altogether without any mods.

That is certainly true, but how many people would touch default settings compared to downloading a mod that does it for them. 

I would argue, it would lose its meaning on 1.20.5(+), tho wouldn't hurt offering it for people who do want it

**Comment by StockiesLad** [comment]
> > Future mc versions?
> 
> I don't think FastLoad is going to be very useful in versions past 1.20.5, since the spawn chunks are now only an area of 3x3 entity ticking chunks by default instead of the 19x19 before, and players can also just disable spawn chunks altogether without any mods.

This is true. Mojang has already implemented to core features of this mod in 1.20.5. Since then, this mod is basically useless. I always had plans to directly build the chunks without using hacks (like skipping LevelLoadingScreen) which would be far more useful for computer with more cores. I also wanted to optimise how and when spawning occurs to speed up the player just getting in the game like it's Minecraft Beta. The reason these things were never worked on was I never had the understanding or motivation to make this happen.

Whether Kirbo ends up picking up or project or not, I will eventually (not for the foreseeable future though) pick this up and implement the features I always had plans for. As for the code mess that is Fastload. Most of the abstraction stuff has already been implemented in a custom api (albeit unfinished) to streamline abstractions. This would make Fastload far easier to work with in the future and maintenance would mainly comprise of just bugfixes. Whilst the back-end will handle all the other stuff.

It's possible in the future that I'll have Kirbo maintain fastload itself, while I work on the back end (the stuff that interests me). Of course, it's important to acknowledge these plans are just **possibilities** and are **not set in stone**.

**Comment by KirboSoftware** [comment]
> Whether Kirbo ends up picking up or project or not

I mean, I can if general interest is there. I just got back from a long break so could pick things back up that I actually want to do. We could discuss this further in DM's @StockiesLad 

I could pick up the project again, it's just requires quite a massive rewrite to make it easier to maintain

**Comment by StockiesLad** [comment]
> > Whether Kirbo ends up picking up or project or not
> 
> I mean, I can if general interest is there. I just got back from a long break so could pick things back up that I actually want to do. We could discuss this further in DM's [@StockiesLad](https://github.com/StockiesLad)
> 
> I could pick up the project again, it's just requires quite a massive rewrite to make it easier to maintain

I'm still on the fence. But if I do get back, don't worry about the pain in the ass rewrite as I'd probably do it myself as I was the guy that made it impossible to read in the first place lmao. The issue is work and study is picking back up so I really don't know what I want to do right now.

If things work out. I'll definitely let you know here and we'll talk about how to make this happen.

**Comment by StockiesLad** [comment]
Thread continued in https://github.com/BumbleSoftware/Fastload/issues/120.
